packages/sanity/src/core/comments/store/useCommentsStore.ts

```diff
@@ -84,7 +83,7 @@ export function useCommentsStore(opts: CommentsStoreOptions): CommentsStoreRetur
   const handleListenerEvent = useCallback(
     async (event: ListenEvent<Record<string, CommentDocument>>) => {
       // Fetch all comments on initial connection
-      if (event.type === 'welcome' && !didInitialFetch.current) {
+      if (event.type === 'welcome') {
```

Comment:
needed to remove it in order for the comments to be refetched on switching perspectives in a document 

https://github.com/user-attachments/assets/a8177816-3d4f-4219-86e6-71a890852d0d

Reply:
Good catch, let's discuss in slack

Reply:
Can you ping the other reviewers when you come to a conclusion on this? 🙂

Reply:
I'll be making this change 🔜